includes/admin/class-wc-admin-status.php

```diff
@@ -51,6 +51,22 @@ public static function status_tools() {
 			} else {
 				echo '<div class="error inline"><p>' . esc_html( $response['message'] ) . '</p></div>';
 			}
+
+			$tool = $tools[ $action ];
```

Comment:
As seen on line 40, there is the potential of the tool not existing for the input action. This should gracefully handle that somehow. The current implementation will cause PHP errors.

Best solution is probably to move this new code inside the `if` block on line 40 and fire the 'woocommerce_system_status_tool_executed ' hook right after the tool is executed. That would also closely match [this one](https://github.com/woocommerce/woocommerce/blob/master/includes/api/class-wc-rest-system-status-tools-controller.php#L259-L287) which only fires when there is a valid tool.